Block/Js.php

```diff
@@ -450,7 +442,13 @@ public function shouldDisableBoltCheckout()
      */
     public function getMagentoStoreId()
     {
-        return (int) ($this->magentoQuote && $this->magentoQuote->getStoreId()) ?
-            $this->magentoQuote->getStoreId() : 0;
+        if ($this->checkoutSession instanceof BackendSessionQuote) {
+            $quote = $this->checkoutSession;
+        } else {
+            $quote = $this->checkoutSession->getQuote();
+        }
+
+        return (int) ($quote && $quote->getStoreId()) ?
```

Comment:
Why casting to int?

Reply:
I mistyped, I wanted to be sure that we return int type but i was mistyped and convert to int only a condition.